### DISCUSSION TITLE
Getting Issues while Integration with MAUI

### DISCUSSION THREAD

**Discussion by piyushvarshi** [TolikPylypchuk/SharpHook#41] [Q&A] [answered]
I am trying to integrate with MAUI and getting this error.
{System.DllNotFoundException: uiohook
at SharpHook.Native.UioHook.SetDispatchProc(DispatchProc dispatchProc, IntPtr userData)
at SharpHook.GlobalHookBase.

Can someone please help me out here in getting this resolved?

**Comment by TolikPylypchuk** [comment, marked as answer]
Hi! Thanks for asking! SharpHook will only work with MAUI on Windows. On Android and iOS it doesn't work because libuiohook, the library which it uses under the hood, doesn't work on these OSes. On macOS MAUI actually uses Mac Catalyst, and apparently libuiohook doesn't support it as well (there's an issue #35 to track that).

**Reply by piyushvarshi** [reply]
Yes trying in Windows only but getting error "Unable to load DLL 'uiohook' or one of its dependencies: The specified module could not be found. "
I thinks I am missing something here on where and how to implement uiohook

**Reply by TolikPylypchuk** [reply]
Now this is a weird thing. The first thing to check is whether the `uiohook.dll` file is in the build output directory (e.g. `bin/Debug`). If no, then it's not copied there from the package correctly. If yes, then it's something else and much worse probably.

If it's not there, then I'll create a MAUI project on my machine and try to reproduce this issue.

**Reply by piyushvarshi** [reply]
Thanks for your prompt support. I was able to resolve the issue. Also looking to implementation of key suppression through this.

**Reply by TolikPylypchuk** [reply]
You can suppress input events in global hook event handlers - the `HookEventArgs` class contains the [`SuppressEvent`](https://sharphook.tolik.io/api/SharpHook.HookEventArgs.html#SharpHook_HookEventArgs_SuppressEvent) property which should be set to `true`. Note, that this works only on Windows and macOS, and requires that the event handler is synchronous, i.e. it will work in `SimpleGlobalHook`, but not `TaskPoolGlobalHook` (since the latter uses asynchronous event handling).

**Comment by TolikPylypchuk** [comment]
Hi @piyushvarshi! FYI, I've just released version 5.0.0 which adds support for .NET MAUI on macOS.
